### DISCUSSION TITLE
API design quirks

### DISCUSSION THREAD

**Discussion by s-zanella** [ethz-spylab/satml-llms-ctf-issues#4] [Questions]
Curious why you chose to have an endpoint to create a new defense **and** a chat that uses it (`/api/v1/chat/create-with-new-defense`), rather than a simpler endpoint to just create a new defense (e.g., `/api/v1/defense/create`) and let participants create chats separately (with `/api/v1/chat/create-with-existing-defense`).

Also, the absence of an endpoint to delete chats is conspicuous (even if you keep them for your own records), although I'd understand if that's a deliberate design choice.

**Comment by dedeswim** [comment]
Hi @s-zanella,
Thanks a lot for your questions and for sharing your feedback.

We are working on implementing endpoints to create a defense independently of a chat and to delete chats!

**Comment by dedeswim** [comment]
Update: we have just pushed a change that adds a new `/api/v1/defense/create` endpoint, and we will soon push one to delete chats
